core/src/main/java/jenkins/model/item_category/NestedProjectsCategory.java

```diff
@@ -0,0 +1,38 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import jenkins.model.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * A generic {@link ItemCategory}.
+ *
+ * This category should be moved to cloudbees-folder-plugin short-term.
+ */
+@Restricted(NoExternalUse.class)
+@Extension(ordinal = -100)
+public class NestedProjectsCategory extends ItemCategory {
+
+    public static final String ID = "nestedprojects";
```

Comment:
Right - so I finally get it.   
This Category is here so that the UI can find the category via extension lookup.  Other plugins can implement the methods without adding `@Override` so that they can compile and run with older cores but when run on 2.x will be seen as TopLevelItems within this category - so long as they return a String with the same contents from `getCategoryId` which until they have a baseline of 2.0+ will have to hard code.

Therefore as no one can access the ID, this can easily be made private until the code is moved to the cloudbees-folders plugin, and there is no requirement for this field to be public!